copier.go

```diff
@@ -0,0 +1,150 @@
+package tests
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"io"
+	"math/rand"
+	"testing"
+
+	"github.com/google/uuid"
+	. "github.com/smartystreets/goconvey/convey"
+
+	"github.com/beyondstorage/go-storage/v4/pkg/randbytes"
+	"github.com/beyondstorage/go-storage/v4/services"
+	"github.com/beyondstorage/go-storage/v4/types"
+)
+
+func TestCopier(t *testing.T, store types.Storager) {
+	Convey("Given a basic Storager", t, func() {
+		Convey("The Storager should implement Copier", func() {
+			_, ok := store.(types.Copier)
+			So(ok, ShouldBeTrue)
+		})
+
+		Convey("When Copy a file", func() {
+			c, _ := store.(types.Copier)
+
+			size := rand.Int63n(4 * 1024 * 1024)
+			r := io.LimitReader(randbytes.NewRand(), size)
+			rMD5Hex := calculateHashFromReader(r)
```

Comment:
The reader could be consumed and not reusable.

How about generating into bytes slice directly? And maybe we can compare the bytes directly (They are small enough).